### PR TITLE
Added in socket io client, added room listener, cleaned up some code

### PR DESCRIPTION
**Pull request by JoshuaWarejko**
Some of the items in this push were just changing some line indentation. Also, not all is complete yet.

### CODE REVIEW

File: frontend/src/app/app.module.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ import { LoginPageComponent } from './pages/login-page/login-page.component';
 import { RegisterPageComponent } from './pages/register-page/register-page.component';
 
 // Providers
-import { AuthService } from './providers/auth-service/auth.service';
+import { AuthService, LobbyService } from './providers';
```
Comment (torch2424) [inline frontend/src/app/app.module.ts]: Woowwwwwww didn't know about dis

**Comment by torch2424** [comment]
Let's pull this later that way we don't spend too much time getting people not to conflict :)

**Comment by JoshuaWarejko** [comment]
Yeah sounds good, sorry some of the files were all over the place in terms of having one push be for one thing haha

File: frontend/src/app/app.module.ts (hunk 1 above)
Comment (JoshuaWarejko) [reply frontend/src/app/app.module.ts]: yeee I believe it is called a barrel index.ts file when you do it that way.

**Pull request merged**